### PR TITLE
Carry 9916: add comment column in docker history command

### LINKED ISSUE #9785: Make comment message to be displayed in docker history command
I raised a discussion here

https://groups.google.com/forum/#!topic/docker-dev/AlbF1ij5oY0

<code>docker commit</code> command accept a <code>-m</code> parameter which allow users to add some messages to their image, just like which people would do while commit their code.

But anyway these messages are not read out when a user issue a <code>docker history</code> command to that image. 

By inspecting into source code, I found these messages were surely stored with images, so I made a patch to fetch them out. 

````patch
root@966e1e937644:/go/src/github.com/docker/docker# git diff 739d917..HEAD
diff --git a/api/client/commands.go b/api/client/commands.go
index a96089b..04d0496 100644
--- a/api/client/commands.go
+++ b/api/client/commands.go
@@ -1036,7 +1036,7 @@ func (cli *DockerCli) CmdHistory(args ...string) error {

 
        w := tabwriter.NewWriter(cli.out, 20, 1, 3, ' ', 0)
        if !*quiet {
-               fmt.Fprintln(w, "IMAGE\tCREATED\tCREATED BY\tSIZE")
+               fmt.Fprintln(w, "IMAGE\tCREATED\tCREATED BY\tSIZE\tCOMMENT")
        }
 
        for _, out := range outs.Data {
@@ -1055,7 +1055,8 @@ func (cli *DockerCli) CmdHistory(args ...string) error {

                        } else {
                                fmt.Fprintf(w, "%s\t", utils.Trunc(out.Get("CreatedBy"), 45))
                        }
-                       fmt.Fprintf(w, "%s\n", units.HumanSize(out.GetInt64("Size")))
+                       fmt.Fprintf(w, "%s\t", units.HumanSize(out.GetInt64("Size")))
+                        fmt.Fprintf(w, "%s\n", out.Get("Comment"))
                } else {
                        if *noTrunc {
                                fmt.Fprintln(w, outID)
diff --git a/graph/history.go b/graph/history.
go
index 2030c4c..853d6c6 100644
--- a/graph/history.go
+++ b/graph/history.go
@@ -36,6 +36,7 @@ func (s *TagStore) CmdHistory(job *engine.Job) engine.Status {

                out.Set("CreatedBy", strings.Join(img.ContainerConfig.Cmd, " "))
                out.SetList("Tags", lookupMap[img.ID])
                out.SetInt64("Size", img.Size)
+                out.Set("Comment", img.Comment)
                outs.Add(out)
                return nil
        })

root@966e1e937644:/go/src/github.com/docker/docker#
````

With this patch you can have a COMMENT column in the output of <code>docker history</code> command.

````shell
root@localhost docker]# ./docker history 2ac9d1098bf1
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
2ac9d1098bf1        12 weeks ago        /bin/bash                                       241.4 MB            Add Git and SVN
88d178232033        12 weeks ago        /bin/bash                                       186 MB              Add JDK1.7
4e5e740790d1        3 months ago        /bin/bash                                       535 MB              
88b42ffd1f7c        5 months ago        /bin/sh -c #(nop) ADD file:1fd8d7f9f6557cafc7   373.7 MB            
c69cab00d6ef        5 months ago        /bin/sh -c #(nop) MAINTAINER Lokesh Mandvekar   0 B                
511136ea3c5a        18 months ago                                                       0 B                 Imported from -


````

### PR DESCRIPTION
**Pull request by tiborvass**
Closes #9785
Closes #9916 

Ping @LK4D4 @icecrime

### CODE REVIEW

**Comment by icecrime** [comment]
Code LGTM

**Comment by cpuguy83** [comment]
Integration test seems a little weird to just test that there is a comment column.

**Comment by cpuguy83** [comment]
That's just a nit, so LGTM moving to docs review.

**Comment by dove-young** [comment]
Sorry for late response. I am not able to access Gmail frequently in China. And thank you very much for carry it. @tiborvass

**Comment by tiborvass** [comment]
@dove-young Oh no worries! Thank you very much for your contribution (should have said that in the description!). It's totally fine and normal to not have time to finish a contribution :)

I kept your signed commits to credit your work :)

**Comment by moxiegirl** [comment]
LGTM

**Comment by moxiegirl** [comment]
ping @fredlf @jamtur01 @SvenDowideit

**Comment by jamtur01** [comment]
The title in the docs "Show history.." isn't grammatically correct. Otherwise LGTM

File: docs/man/docker-history.1.md [hunk 1]
```diff
@@ -27,10 +27,20 @@ Show the history of when and how an image was created.
...
 # EXAMPLES
     $ docker history fedora
-    IMAGE          CREATED          CREATED BY                                      SIZE
+    IMAGE          CREATED          CREATED BY                                      SIZE                COMMENT
     105182bb5e8b   5 days ago       /bin/sh -c #(nop) ADD file:71356d2ad59aa3119d   372.7 MB
     73bd853d2ea5   13 days ago      /bin/sh -c #(nop) MAINTAINER Lokesh Mandvekar   0 B
-    511136ea3c5a   10 months ago                                                    0 B
+    511136ea3c5a   10 months ago                                                    0 B                 Imported from -
+
+## Show the history of images created through docker commit command
```
Comment (moxiegirl) [inline docs/man/docker-history.1.md]: Display comments in the image history

**Comment by fredlf** [comment]
Agreed, need to fix that header. Otherwise, LGTM.

**Comment by tiborvass** [comment]
@fredlf @moxiegirl fixed

**Comment by jamtur01** [comment]
Docs LGTM

**Pull request merged**